File: easy/compare_points.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    chomp($line);
    my ( $x1, $y1, $x2, $y2 ) = split( / /, $line );
    print(
          ( $x1 == $x2 )
        ? ( $y1 == $y2 ? "here\n" : ( $y1 < $y2 ? "N\n" : "S\n" ) )
        : (
            $y1 == $y2
            ? ( $x1 < $x2 ? "E\n" : "W\n" )
            : (
                $x1 < $x2
                ? ( $y1 < $y2 ? "NE\n" : "SE\n" )
                : ( $y1 < $y2 ? "NW\n" : "SW\n" )
            )
        )
    );
}
close(INFILE);

File: easy/compare_points.rb
File.open(ARGV[0]).each_line do |line|
  s = line.split.map(&:to_i)
  if s[0] == s[2]
    if s[1] == s[3]
      puts 'here'
    else
      puts s[1] < s[3] ? 'N' : 'S'
    end
  elsif s[1] == s[3]
    puts s[0] < s[2] ? 'E' : 'W'
  elsif s[0] < s[2]
    puts s[1] < s[3] ? 'NE' : 'SE'
  else
    puts s[1] < s[3] ? 'NW' : 'SW'
  end
end
